inc/class-wpseo-image-utils.php

```diff
@@ -0,0 +1,270 @@
+
+		if ( function_exists( 'wpcom_vip_attachment_url_to_postid' ) ) {
+			// @codeCoverageIgnoreStart -- we can't test this properly.
+			return (int) wpcom_vip_attachment_url_to_postid( $url );
+			// @codeCoverageIgnoreEnd -- The rest we _can_ test.
+		}
+
+		// phpcs:ignore WordPress.VIP.RestrictedFunctions -- We use the WP COM version if we can, see above.
+		return (int) attachment_url_to_postid( $url );
+	}
+
+	/**
+	 * Finds an image size that is under 2 MB and thus viable to use for social sharing.
+	 *
+	 * @param int $attachment_id The attachment ID.
+	 *
+	 * @return array|false Image array on success, false on failure.
+	 */
+	public static function find_correct_image_size( $attachment_id ) {
+		foreach ( self::get_image_sizes() as $size ) {
+			$image = self::check_image_by_size( $attachment_id, $size );
+			if ( $image !== false ) {
+				return $image;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Enrich the image data to use elsewhere.
+	 *
+	 * @param array $image         Image array with URL and metadata.
+	 * @param int   $attachment_id Attachment ID.
+	 *
+	 * @return array $image {
+	 *     Array of image data
+	 *
+	 *     @type string $alt    Image's alt text.
+	 *     @type string $alt    Image's alt text.
+	 *     @type int    $width  Width of image.
+	 *     @type int    $height Height of image.
+	 *     @type string $type   Image's MIME type.
+	 *     @type string $url    Image's URL.
+	 * }
+	 */
+	public static function complete_image_data( $image, $attachment_id ) {
+		$image['id']     = $attachment_id;
+		$image['alt']    = self::get_image_alt_tag( $attachment_id );
+		$image['pixels'] = ( $image['width'] * $image['height'] );
+
+		if ( ! isset( $image['type'] ) ) {
+			$image['type'] = get_post_mime_type( $attachment_id );
+		}
+		// Keep only the keys we need, and nothing else.
+		$image = array_intersect_key( $image, array_flip( array( 'id', 'alt', 'path', 'width', 'height', 'pixels', 'type', 'size', 'url' ) ) );
+		return $image;
+	}
+
+	/**
+	 * Check original size of image. If original image is too small, return false, else return true.
+	 *
+	 * @param int   $attachment_id The attachment ID to check the size of.
+	 * @param array $params {
+	 *    The parameters to check against.
+	 *
+	 *    @type int    $min_width     Minimum width of image.
+	 *    @type int    $max_width     Maximum width of image.
+	 *    @type int    $min_height    Minimum height of image.
+	 *    @type int    $max_height    Maximum height of image.
+	 *    @type int    $min_ratio     Minimum aspect ratio of image.
+	 *    @type int    $max_ratio     Maximum aspect ratio of image.
+	 * }
+	 *
+	 * @return bool Whether an image is fit for display or not.
+	 */
+	public static function check_image_measurements( $attachment_id, $params ) {
+		$img_data = wp_get_attachment_metadata( $attachment_id );
+		if ( ! isset( $img_data['width'] ) || ! isset( $img_data['height'] ) ) {
```

Comment:
If changing this to ` ! empty` we can avoid `division by zero` situations in the code below.